quic/s2n-quic-core/src/datagram/traits.rs

```diff
@@ -36,9 +39,20 @@ impl Default for ConnectionInfo {
     }
 }
 
+/// PreConnectionInfo will contain information needed to determine how
+/// large of a datagram a provider is willing to accept
+#[non_exhaustive]
+pub struct PreConnectionInfo(());
+
+impl PreConnectionInfo {
+    pub fn new() -> Self {
```

Comment:
this should not be public

Reply:
It needs to be so it can be constructed in the transport crate. Instead, I would add doc(hidden) so it's not documented

```suggestion
    #[doc(hidden)]
    pub fn new() -> Self {
```

Reply:
You can't create non_exhaustive structs outside of the crate that defines them. Otherwise you would break callers anytime new fields are added